community/google-summer-of-code/reports/2017/lambda-Support-in-debugger-expression-evaluator.md

```diff
@@ -0,0 +1,135 @@
+---
+title: "Lambda Support in debugger expression evaluator (Haruka Matsumoto)"
+---
+
+Author: Haruka Matsumoto
+
+# Abstract
+
+My project is lambda support in the MonoDevelop debugger's expression evaluator, used in the Immediate pad.
+
+# Code
+
+All of my work is covered by a single pull request: [here](https://github.com/mono/debugger-libs/pull/142#pull)
+
+# Summary of the implementation
+
+![Processing flow chart](/images/gsoc-2017-lambda-support-in-debugger expression-evaluator.png "Processing flow chart")
+
+*Fig 1. Processing flow chart*
+
+The debugger works using an interface [SDWP](http://www.mono-project.com/docs/advanced/runtime/docs/soft-debugger-wire-format/) exposed by Mono runtime. Using this protocol, the debugger controls debuggees and queries information of the debuggees like address of objects, type or methods. The debugger's evaluator usually works as an interpreter, which evaluates every part of expressions immediately. However, lambda support can not be realized in this way. We don't know when the debuggees invoke lambdas. The code of a lambda needs to stay invocable in the debuggees. This is how lambda expressions are different from other ones for the debugger's evaluator.
```

Comment:
We are not using SDWP acronym

Reply:
also please remove the `http://www.mono-project.com` prefix from the link (the link can be relative).

Reply:
Done.